apps/explorer/src/components/owned-objects/views/FieldItem.tsx

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { type SuiMoveNormalizedType } from '@mysten/sui.js';
+
+import { SyntaxHighlighter } from '~/components/SyntaxHighlighter';
+import { getFieldTypeValue } from '~/components/owned-objects/utils';
+import { AddressLink, ObjectLink } from '~/ui/InternalLink';
+import { Link } from '~/ui/Link';
+import { Text } from '~/ui/Text';
+
+interface FieldItemProps {
+    value: string | number | object | boolean;
+    type: SuiMoveNormalizedType;
+    truncate?: boolean;
+}
+
+const TYPE_ADDRESS = 'Address';
+const TYPE_URL = '0x2::url::Url';
+const TYPE_OBJECT_ID = ['0x2::object::UID', '0x2::object::ID'];
+
+export function FieldItem({ value, type, truncate = false }: FieldItemProps) {
+    // for object types, use SyntaxHighlighter
+    if (typeof value === 'object') {
+        return (
+            <SyntaxHighlighter
+                code={JSON.stringify(value, null, 2)}
+                language="json"
+            />
+        );
+    }
+
+    const { displayName, normalizedType } = getFieldTypeValue(type);
+
+    if (displayName === TYPE_ADDRESS) {
```

Comment:
This is actually _slightly_ wrong because this means that `0xmyPackage::whatever::Address` would display as an address right? What you probably want to do is to make the `normalizedType` _always_ populated and just populate it with the same value as the display name in the case that it's just a built-in type. That'd avoid this problem.

Reply:
Thanks for pointing this out, I made the change.